components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/DefaultOAuth2TokenValidator.java

```diff
@@ -42,38 +52,82 @@ public boolean validateAccessDelegation(OAuth2TokenValidationMessageContext mess
         Set<OAuth2ScopeValidator> oAuth2ScopeValidators = OAuthServerConfiguration.getInstance()
                 .getOAuth2ScopeValidators();
-        for (OAuth2ScopeValidator validator: oAuth2ScopeValidators) {
-
-            if (validator != null && validator.canHandle(messageContext)) {
-                String resource = null;
-                if (messageContext.getRequestDTO().getContext() != null) {
-                    //Iterate the array of context params to find the 'resource' context param.
-                    for (OAuth2TokenValidationRequestDTO.TokenValidationContextParam resourceParam :
-                            messageContext.getRequestDTO().getContext()) {
-                        //If the context param is the resource that is being accessed
-                        if (resourceParam != null && "resource".equals(resourceParam.getKey())) {
-                            resource = resourceParam.getValue();
-                            break;
-                        }
-                    }
+        ArrayList<String> appScopeValidators = new ArrayList<>(Arrays.asList(scopeValidators));
+        for (OAuth2ScopeValidator validator : oAuth2ScopeValidators) {
+            if (validator != null && appScopeValidators.contains(validator.getClass().getSimpleName())
+                    && validator.canHandle(messageContext)) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Validating scope of token %s using %s", accessTokenDO.getTokenId(),
+                            validator.getClass().getName()));
                 }
-                boolean isValid = validator.validateScope((AccessTokenDO) messageContext.getProperty
-                        (ACCESS_TOKEN_DO), resource);
-
+                boolean isValid = validator.validateScope(accessTokenDO, resource);
+                appScopeValidators.remove(validator.getClass().getSimpleName());
                 if (!isValid) {
                     return false;
                 }
             }
         }
+        if (!appScopeValidators.isEmpty()) {
+            throw new IdentityOAuth2Exception(String.format("The scope validators %s registered for application " +
+                    "%s@%s are not found in the server configuration ", StringUtils.join(appScopeValidators, ", "),
+                    app.getApplicationName(), OAuth2Util.getTenantDomainOfOauthApp(app)));
+        }
         return true;
     }
 
+    /**
+     *  Extract the resource from the access token validation request message
+     * @param messageContext Message context of the token validation request
+     * @return resource
+     */
+    private String getResourceFromMessageContext(OAuth2TokenValidationMessageContext messageContext) {
+        String resource = null;
+        if (messageContext.getRequestDTO().getContext() != null) {
+            //Iterate the array of context params to find the 'resource' context param.
```

Comment:
Add a space before starting the comment. Correct in other places as well.

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/942/commits/ca4690d734709b108ef2df33a69ae3c9242febc8